modules/querying/pages/func/query-user-defined-functions.adoc

```diff
@@ -1,46 +1,67 @@
 = Query User-Defined Functions
 :pp: {plus}{plus}
 
-In GSQL, users can supplement the language by defining their own query user-defined functions (query UDF). Query UDFs can be called in queries and subqueries to perform a set of defined actions and return a value like the built-in functions.
+In GSQL, users can supplement the language by defining their own query user-defined functions (query UDF) in C{pp}. Query UDFs can be called in queries and subqueries to perform a set of defined actions and return a value like the built-in functions.
 
-This page introduces the process to define a query UDF. Once defined, the new functions will be added into GSQL automatically next time GSQL is executed.
+This page introduces the process to define a query UDF. Once defined, the new functions are added into GSQL automatically the next time GSQL is executed.
+
+Query UDFs are user-editable in the form of an `.hpp` file locally on your machine. These get uploaded and interpreted by the server through a GSQL command. Creating, editing, or removing Query UDFs is done by downloading them from the server, editing the file, and reuploading.
 
 == Define a query UDF
 
 Below are the steps to add a Query UDF to GSQL:
 
 === Step 1: Download current query UDF file
 
-Use the `GET ExprFunctions` command in GSQL to download the current UDF file to any location on your machine. The file and the directores will be created if they do not exist, and the file must end with the file extention `.hpp`:
+Use the `GET ExprFunctions` command in GSQL to download the current list of functions into a local file. The path can be absolute or relative to your current directory, but the file extension must be `.hpp`:
```

Comment:
"to download the current list of functions into a local file": I'm not sure this is correct either. The command does not download a list.  It downloads a file which contains a _set_ of UDFs.  "If no UDFs have been defined yet, then the file will be empty." <-- Is this true?